public/js/actions/sources.js

```diff
@@ -36,9 +40,44 @@ function queueSourcesDispatch(dispatch) {
  * Handler for the debugger client's unsolicited newSource notification.
  */
 function newSource(source) {
-  return ({ dispatch }) => {
-    if (isEnabled("features.sourceMaps") && source.sourceMapUrl) {
-      networkRequest(source.sourceMapUrl).then(res => {
+  return ({ dispatch, getState }) => {
+    if (isEnabled("features.sourceMaps") && source.sourceMapURL) {
+      const tab = getSelectedTab(getState());
+      const sourceMapURL = `${tab.get("url")}/${source.sourceMapURL}`;
+      networkRequest(sourceMapURL).then(res => {
+        Task.spawn(function* () {
+          yield dispatch(loadSourceText(source));
+          const sourceText = getSourceText(getState(), source.id)
```

Comment:
We can't eagerly load in the source text for sources when they are introduced. This would create a huge amount of protocol traffic. It's already slow to transfer very large sources, and we don't want to do that at startup for every source.

The `SourceMapConsumer` should have a `sources` property which already lists the sources that the sourcemap represents. You should be able to dispatch new sources based on that, without needing any of the source contents until someone clicks on a source.